### PR TITLE
Make `x clippy` download and use beta clippy

### PR DESCRIPTION
**Pull request by albertlarsan68**
Take-over of #106394: see this PR for all of the context.

r? @flip1995 
cc @jyn514

### CODE REVIEW

**Comment by jyn514** [comment]
See https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Overriding.20RUSTC_WORKSPACE_WRAPPER.20with.20RUSTC/near/325529313 for the current status - the clippy changes here aren't something we want to land as-is.

File: src/bootstrap/bin/rustc.rs [hunk 1]
```diff
@@ -37,14 +37,25 @@ fn main() {
...
+    }
+
     // Use a different compiler for build scripts, since there may not yet be a
     // libstd for the real compiler to use. However, if Cargo is attempting to
     // determine the version of the compiler, the real compiler needs to be
     // used. Currently, these two states are differentiated based on whether
     // --target and -vV is/isn't passed.
     let (rustc, libdir) = if target.is_none() && version.is_none() {
+        if verbose > 1 {
+            eprintln!("Using snapshot complier");
```
Comment (kadiwa4) [inline src/bootstrap/bin/rustc.rs:52]: *compiler, same below

**Comment by anden3** [comment]
@kadiwa4 Just want to ping you as part of the triage procedure to let you know that this PR has merge conflicts :)

**Comment by anden3** [comment]
@albertlarsan68 Just want to ping you as part of the triage procedure to let you know that this PR has merge conflicts :)

**Comment by albertlarsan68** [comment]
I know, I need to work more on this PR.
I might get to it later :)

**Comment by JohnCSimon** [comment]
@albertlarsan68 
ping from triage - can you post your status on this PR? There hasn't been an update in a few months. Thanks!

**Comment by albertlarsan68** [comment]
I'll be unavailable for 1 to 2 months to work on it, so I'll mark it as blocked on me.
Concerning the status, it is kinda stuck in a limbo between multiple implementation difficulties. If someone wants to continue this, they can contact me on Zulip and I'll try to answer.

**Pull request closed without merging**